apps/dashboard/src/main/java/com/akto/action/testing_issues/IssuesAction.java

```diff
@@ -225,10 +225,29 @@ public static BasicDBObject createSubcategoriesInfoObj(TestConfig testConfig) {
     }
 
     private boolean fetchOnlyActive;
+    private String mode;
 
     public String fetchAllSubCategories() {
 
-        Map<String, TestConfig> testConfigMap  = YamlTemplateDao.instance.fetchTestConfigMap(true, fetchOnlyActive);
+        boolean includeYamlContent = false;
+
+        switch (mode) {
+            case "RunTests":
+                categories = GlobalEnums.TestCategory.values();
+                break;
+            case "TestEditor":
+                includeYamlContent = true;
+                vulnerableRequests = VulnerableRequestForTemplateDao.instance.findAll(Filters.empty());
+                break;
+            case "Dashboard":
+                testSourceConfigs = TestSourceConfigsDao.instance.findAll(Filters.empty());
+                break;
+            default:
+                return "ERROR: Invalid mode";
+        }
+
+        Map<String, TestConfig> testConfigMap = YamlTemplateDao.instance.fetchTestConfigMap(includeYamlContent,
```

Comment:
See YamlTemplateDao.java, there we parse the template before using includeYamlContent's value, this means we will just not send it to the UI but it will still parse it, thus modify that file also